src/hooks/useAllCities.js

```diff
@@ -0,0 +1,39 @@
+import { useMemo } from 'react';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import fetchCities from '../services/fetchCitiesServise';
+import transformCityItem from '../data/transformCityItem';
+
+const ALL_CITIES_QUERY_KEY = 'allCities';
+
+const useAllCities = () => {
+  const { data, error, isLoading, fetchNextPage, hasNextPage } =
+    useInfiniteQuery({
+      queryKey: [ALL_CITIES_QUERY_KEY],
+      queryFn: ({ pageParam = 1 }) =>
+        fetchCities({
+          page: pageParam,
+          pageSize: 50,
+          sortOrder: 'NAME_ASC',
+          onlyWithGuides: false,
+        }),
+
+      getNextPageParam: (lastPage, pages) => {
+        return lastPage.length > 0 ? pages.length + 1 : undefined;
```

Comment:
Your calculation is wrong, according to your logic you are making an extra request to figure out if there is a next page

You can use `hasMore` field from the response to understand if you have the next page or no. This will help to avoid sending extra query